dockersetup.py

```diff
@@ -0,0 +1,904 @@
+#!/usr/bin/env python3
+
+# Layer index Docker setup script
+#
+# Copyright (C) 2018 Intel Corporation
+# Author: Amber Elliot <amber.n.elliot@intel.com>
+#
+# Licensed under the MIT license, see COPYING.MIT for details
+#
+# SPDX-License-Identifier: MIT
+
+# This script will make a cluster of 5 containers:
+#
+#  - layersapp: the application
+#  - layersdb: the database
+#  - layersweb: NGINX web server (as a proxy and for serving static content)
+#  - layerscelery: Celery (for running background jobs)
+#  - layersrabbit: RabbitMQ (required by Celery)
+#
+# It will build and run these containers and set up the database.
+
+import sys
+
+min_version = (3, 4, 3)
+if sys.version_info < min_version:
+    sys.stderr.write('Sorry, python version %d.%d.%d or later is required\n' % min_version)
+    sys.exit(1)
+
+import os
+import argparse
+import re
+import subprocess
+import time
+import random
+import shutil
+import tempfile
+from shlex import quote
+
+def get_args():
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 24. By comparison, 99% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 119-132 into a separate function.